---
title: "10 Most importante taxonomy 16S"
author: "JRL"
date: "15/06/2020"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(purrr)
library(openxlsx)
library(ggplot2)
library(tidyverse)
```

```{r cars}
(df <- as.data.frame(read.xlsx("~/Documentos/SupAgro - Facul/JRL/microbiome_soil-master/microbiome_soil-master/sequences_processing/16S/16S_results/16S_rarefied_table.xlsx")))
# Tidy the names from the imported table (no spaces, no number in taxonomic columns)
df$Phylum <- gsub(" ","", df$Phylum)
df$Order <- gsub(" ", "", df$Order)
df$Family <- gsub(" ", "", df$Family)
df$Gender <- gsub(" ", "", df$Gender)
df$Gender <- gsub("\\d","", df$Gender)
df$Gender <- gsub("([[:punct:]]\\d)","", df$Gender)

```
```{r percentages_per_read}
## PHYLUM
df_phylum <- na.omit(df[,-c(1,3:7)])
ad0 <- as.vector(NULL)
for (s in 2:ncol(df_phylum)) {
  sum_phylum <- tapply(df_phylum[,s], as.factor(df_phylum$Phylum), sum)
   ad <- tapply(df_phylum[,s], as.factor(df_phylum$Phylum), sum)*100 / sum(df_phylum[,s])
   ad0 <- cbind(ad0,ad)
}
colnames(ad0)<-names(df_phylum)[2:ncol(df_phylum)]
#Percentage of unidentified phylum
pct_p <- (sum_phylum["unidentified"]/sum(sum_phylum))*100

#Percentage in sequences of
pct_actino <- (sum_phylum["Actinobacteriota"]/sum(sum_phylum))*100
pct_proteo <- (sum_phylum["Proteobacteria"]/sum(sum_phylum))*100
pct_bact <- (sum_phylum["Bacteroidota"]/sum(sum_phylum))*100
pct_chl <- (sum_phylum["Chloroflexi"]/sum(sum_phylum))*100  
pct_cren <- (sum_phylum["Crenarchaeota"]/sum(sum_phylum))*100
pct_gemma <- (sum_phylum["Gemmatimonadota"]/sum(sum_phylum))*100
pct_planc <- (sum_phylum["Planctomycetota"]/sum(sum_phylum))*100
pct_acido <- (sum_phylum["Acidobacteriota"]/sum(sum_phylum))*100
pct_cyano <- (sum_phylum["Cyanobacteria"]/sum(sum_phylum))*100
pct_verru <- (sum_phylum["Verrucomicrobiota"]/sum(sum_phylum))*100

#Ten most important
rowSums(ad0) > 100
table(rowSums(ad0) > 100)
```

```{r percentages_per_read}
##ORDER
df_order <- na.omit(df[,-c(1:3,5:7)])
ad0_o <- as.vector(NULL)
for (s in 2:ncol(df_order)) {
  sum_order <- tapply(df_order[,s], as.factor(df_order$Order), sum)
   ad_o <- tapply(df_order[,s], as.factor(df_order$Order), sum)*100 / sum(df_order[,s])
   ad0_o <- cbind(ad0_o,ad_o)
}
colnames(ad0_o)<-names(df_order)[2:ncol(df_order)]

#Pct unidentified order
pct_o <- (sum_order["unidentified"]/sum(sum_order))*100

#Ten most important 
 rowSums(ad0_o) > 115
 table(rowSums(ad0_o) > 115)
 
 #Percentage in sequences of
pct_bulk <- (sum_phylum["Burkholderiales"]/sum(sum_phylum,na.rm = TRUE))*100
pct_cyto <- (sum_phylum["Cytophagales"]/sum(sum_phylum))*100
pct_gemmo <- (sum_phylum["Gemmatimonadales"]/sum(sum_phylum))*100
pct_micro <- (sum_phylum["Micrococcales"]/sum(sum_phylum))*100  
pct_prop <- (sum_phylum["Propionibacteriales"]/sum(sum_phylum))*100
pct_gemma <- (sum_phylum["Xanthomonadales"]/sum(sum_phylum))*100
pct_vici <- (sum_phylum["Vicinamibacterales"]/sum(sum_phylum))*100
pct_tepi <- (sum_phylum["Tepidisphaerales"]/sum(sum_phylum))*100
pct_nitro <- (sum_phylum["Nitrososphaerales"]/sum(sum_phylum))*100
pct_chiti <- (sum_phylum["Chitinophagales"]/sum(sum_phylum))*100

```

```{r percentages_per_read}
##FAMILY
df_family <- na.omit(df[,-c(1:4,6:7)])
ad0_f <- as.vector(NULL)
for (s in 2:ncol(df_family)) {
   ad_f <- tapply(df_family[,s], as.factor(df_family$Family), sum)*100 / sum(df_family[,s])
   ad0_f <- cbind(ad0_f,ad_f)
}
colnames(ad0_f)<-names(df_family)[2:ncol(df_family)]

 #Percentage in sequences of
pct_chi <- (sum_phylum["Chitinophagaceae"]/sum(sum_phylum))*100
pct_hyme <- (sum_phylum["Hymenobacteraceae"]/sum(sum_phylum))*100
pct_mic <- (sum_phylum["Microscillaceae"]/sum(sum_phylum))*100
pct_xant <- (sum_phylum["Xanthomonadaceae"]/sum(sum_phylum))*100  
pct_WD <- (sum_phylum["WD2101_soil_group"]/sum(sum_phylum))*100
pct_nit <- (sum_phylum["Nitrososphaeraceae"]/sum(sum_phylum))*100
pct_gem <- (sum_phylum["Gemmatimonadaceae"]/sum(sum_phylum))*100
pct_nocar <- (sum_phylum["Nocardioidaceae"]/sum(sum_phylum))*100
pct_microco <- (sum_phylum["Micrococcaceae"]/sum(sum_phylum))*100
pct_unc <- (sum_phylum["uncultured"]/sum(sum_phylum))*100

#Ten most important 
 rowSums(ad0_f) > 105
 table(rowSums(ad0_f) > 105)

```

```{r percentages_per_read}
##GENDER
df_gender <- na.omit(df[,-c(1:5,7)])
ad0_g <- as.vector(NULL)
for (s in 2:ncol(df_gender)) {
  sum_gender <- tapply(df_gender[,s], as.factor(df_gender$Gender), sum)
  ad_g <- tapply(df_gender[,s], as.factor(df_gender$Gender), sum)*100 / sum(df_gender[,s])
  ad0_g <- cbind(ad0_g,ad_g)
}
colnames(ad0_g)<-names(df_gender)[2:ncol(df_gender)]
#Pct unidentified gender
pct_g <- (sum_gender["unidentified"]/sum(sum_gender))*100

#Ten most important 
 rowSums(ad0_f) > 105
 table(rowSums(ad0_f) > 105)
```

```{r percentage_per_otu}
#Phylum
df_phylum_sorted <- filter(df_phylum, Phylum == "unidentified")
(pct <- (nrow(df_phylum_sorted)/nrow(df))*100)
#Gender
df_gender_sorted <- filter(df_gender, Gender == "unidentified")
(pct <- (nrow(df_gender_sorted)/nrow(df))*100) 
#Order
df_order_sorted <- filter(df_order, Order == "unidentified")
(pct <- (nrow(df_order_sorted)/nrow(df))*100)
```