---
title: "10 Most importante taxonomy 16S"
author: "JRL"
date: "15/06/2020"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(purrr)
library(openxlsx)
library(ggplot2)
library(tidyverse)
```

```{r cars}
(df <- as.data.frame(read.xlsx("~/Documentos/SupAgro - Facul/JRL/microbiome_soil-master/microbiome_soil-master/sequences_processing/16S/16S_results/16S_rarefied_table.xlsx")))
# Tidy the names from the imported table (no spaces, no number in taxonomic columns)
df$Phylum <- gsub(" ","", df$Phylum)
df$Order <- gsub(" ", "", df$Order)
df$Family <- gsub(" ", "", df$Family)
df$Gender <- gsub(" ", "", df$Gender)
df$Gender <- gsub("\\d","", df$Gender)
df$Gender <- gsub("([[:punct:]]\\d)","", df$Gender)

```

```{r percentages_per_read}
## PHYLUM
df_phylum <- na.omit(df[,-c(1,3:7)])
ad0 <- as.vector(NULL)
for (s in 2:ncol(df_phylum)) {
  sum_phylum <- tapply(df_phylum[,s], as.factor(df_phylum$Phylum), sum)
   ad <- tapply(df_phylum[,s], as.factor(df_phylum$Phylum), sum)*100 / sum(df_phylum[,s])
   ad0 <- cbind(ad0,ad)
}
colnames(ad0)<-names(df_phylum)[2:ncol(df_phylum)]
#Percentage of unidentified phylum
pct_p <- (sum_phylum["unidentified"]/sum(sum_phylum))*100

#Percentage in sequences of
pct_actino <- (sum_phylum["Actinobacteriota"]/sum(sum_phylum))*100
pct_proteo <- (sum_phylum["Proteobacteria"]/sum(sum_phylum))*100
pct_bact <- (sum_phylum["Bacteroidota"]/sum(sum_phylum))*100
pct_chl <- (sum_phylum["Chloroflexi"]/sum(sum_phylum))*100  
pct_cren <- (sum_phylum["Crenarchaeota"]/sum(sum_phylum))*100
pct_gemma <- (sum_phylum["Gemmatimonadota"]/sum(sum_phylum))*100
pct_planc <- (sum_phylum["Planctomycetota"]/sum(sum_phylum))*100
pct_acido <- (sum_phylum["Acidobacteriota"]/sum(sum_phylum))*100
pct_cyano <- (sum_phylum["Cyanobacteria"]/sum(sum_phylum))*100
pct_verru <- (sum_phylum["Verrucomicrobiota"]/sum(sum_phylum))*100

#Ten most important
rowSums(ad0) > 100
table(rowSums(ad0) > 100)
```

```{r percentages_per_read}
##ORDER
df_order <- na.omit(df[,-c(1:3,5:7)])
ad0_o <- as.vector(NULL)
for (s in 2:ncol(df_order)) {
  sum_order <- tapply(df_order[,s], as.factor(df_order$Order), sum)
   ad_o <- tapply(df_order[,s], as.factor(df_order$Order), sum)*100 / sum(df_order[,s])
   ad0_o <- cbind(ad0_o,ad_o)
}
colnames(ad0_o)<-names(df_order)[2:ncol(df_order)]

#Pct unidentified order
pct_o <- (sum_order["unidentified"]/sum(sum_order))*100

#Ten most important 
 rowSums(ad0_o) > 115
 table(rowSums(ad0_o) > 115)
 
 #Percentage in sequences of
pct_bulk <- (sum_order["Burkholderiales"]/sum(sum_order))*100
pct_cyto <- (sum_order["Cytophagales"]/sum(sum_order))*100
pct_gemmo <- (sum_order["Gemmatimonadales"]/sum(sum_order))*100
pct_micro <- (sum_order["Micrococcales"]/sum(sum_order))*100  
pct_prop <- (sum_order["Propionibacteriales"]/sum(sum_order))*100
pct_xanth <- (sum_order["Xanthomonadales"]/sum(sum_order))*100
pct_vici <- (sum_order["Vicinamibacterales"]/sum(sum_order))*100
pct_tepi <- (sum_order["Tepidisphaerales"]/sum(sum_order))*100
pct_nitro <- (sum_order["Nitrososphaerales"]/sum(sum_order))*100
pct_chiti <- (sum_order["Chitinophagales"]/sum(sum_order))*100

```

```{r percentages_per_read}
##FAMILY
df_family <- na.omit(df[,-c(1:4,6:7)])
ad0_f <- as.vector(NULL)
for (s in 2:ncol(df_family)) {
  sum_family <- tapply(df_family[,s], as.factor(df_family$Family), sum)
  ad_f <- tapply(df_family[,s], as.factor(df_family$Family), sum)*100 / sum(df_family[,s])
  ad0_f <- cbind(ad0_f,ad_f)
}
colnames(ad0_f)<-names(df_family)[2:ncol(df_family)]

#Ten most important 
 rowSums(ad0_f) > 105
 table(rowSums(ad0_f) > 105)
 
 #Percentage in sequences of
pct_chi <- (sum_family["Chitinophagaceae"]/sum(sum_family))*100
pct_hyme <- (sum_family["Hymenobacteraceae"]/sum(sum_family))*100
pct_mic <- (sum_family["Microscillaceae"]/sum(sum_family))*100
pct_xant <- (sum_family["Xanthomonadaceae"]/sum(sum_family))*100  
pct_WD <- (sum_family["WD2101_soil_group"]/sum(sum_family))*100
pct_nit <- (sum_family["Nitrososphaeraceae"]/sum(sum_family))*100
pct_gem <- (sum_family["Gemmatimonadaceae"]/sum(sum_family))*100
pct_nocar <- (sum_family["Nocardioidaceae"]/sum(sum_family))*100
pct_microco <- (sum_family["Micrococcaceae"]/sum(sum_family))*100
pct_unc <- (sum_family["uncultured"]/sum(sum_family))*100
```

```{r percentages_per_read}
##GENDER
df_gender <- na.omit(df[,-c(1:5,7)])
ad0_g <- as.vector(NULL)
for (s in 2:ncol(df_gender)) {
  sum_gender <- tapply(df_gender[,s], as.factor(df_gender$Gender), sum)
  ad_g <- tapply(df_gender[,s], as.factor(df_gender$Gender), sum)*100 / sum(df_gender[,s])
  ad0_g <- cbind(ad0_g,ad_g)
}
colnames(ad0_g)<-names(df_gender)[2:ncol(df_gender)]
#Pct unidentified gender
pct_g <- (sum_gender["unidentified"]/sum(sum_gender))*100


#Ten most important 
 rowSums(ad0_g) > 60
 table(rowSums(ad0_g) > 60)
 
#Percentage in sequences of Gender
pct_adha <- (sum_gender["Adhaeribacter"]/sum(sum_gender))*100
pct_flavo <- (sum_gender["Flavobacterium"]/sum(sum_gender))*100
pct_ni <- (sum_gender["Nitrososphaeraceae"]/sum(sum_gender))*100
pct_soli <- (sum_gender["Solirubrobacter"]/sum(sum_gender))*100  
pct_un <- (sum_gender["unidentified"]/sum(sum_gender))*100
pct_wd_ <- (sum_gender["WD_soil_group"]/sum(sum_gender))*100
pct_uncu <- (sum_gender["uncultured"]/sum(sum_gender))*100
pct_pse <- (sum_gender["Pseudomonas"]/sum(sum_gender))*100
pct_mass <- (sum_gender["Massilia"]/sum(sum_gender))*100
pct_chlo <- (sum_gender["Chloroplast"]/sum(sum_gender))*100
```

```{r percentages_per_read}
##CLASS
df_class <- na.omit(df[,-c(1:2,4:7)])
ad0_c <- as.vector(NULL)
for (s in 2:ncol(df_class)) {
  sum_class <- tapply(df_class[,s], as.factor(df_class$Class), sum)
  ad_c <- tapply(df_class[,s], as.factor(df_class$Class), sum)*100 / sum(df_class[,s])
  ad0_c <- cbind(ad0_c,ad_c)
}
colnames(ad0_c)<-names(df_class)[2:ncol(df_class)]
#Pct unidentified class
pct_c <- (sum_class["unidentified"]/sum(sum_class))*100

#Ten most important 
 rowSums(ad0_c) > 150
 table(rowSums(ad0_c) > 150)
 
 #Percentage in sequences of Class
pct_act <- (sum_class["Actinobacteria"]/sum(sum_class))*100
pct_alpha <- (sum_class["Alphaproteobacteria"]/sum(sum_class))*100
pct_bac <- (sum_class["Bacteroidia"]/sum(sum_class))*100
pct_ch <- (sum_class["Chloroflexia"]/sum(sum_class))*100  
pct_gamma <- (sum_class["Gammaproteobacteria"]/sum(sum_class))*100
pct_nitroso <- (sum_class["Nitrososphaeria"]/sum(sum_class))*100
pct_phy <- (sum_class["Phycisphaerae"]/sum(sum_class))*100
pct_plant <- (sum_class["Planctomycetes"]/sum(sum_class))*100
pct_ther <- (sum_class["Thermoleophilia"]/sum(sum_class))*100
pct_ver <- (sum_class["Verrucomicrobiae"]/sum(sum_class))*100
```

```{r percentages_per_read}
##SPECIES
df_species <- na.omit(df[,-c(1:6)])
ad0_s <- as.vector(NULL)
for (s in 2:ncol(df_species)) {
  sum_species <- tapply(df_species[,s], as.factor(df_species$Species), sum)
  ad_s <- tapply(df_species[,s], as.factor(df_species$Species), sum)*100 / sum(df_species[,s])
  ad0_s <- cbind(ad0_s,ad_s)
}
colnames(ad0_s)<-names(df_species)[2:ncol(df_species)]
#Pct unidentified class
pct_s <- (sum_species["unidentified"]/sum(sum_species))*100

#Ten most important 
 rowSums(ad0_s) > 28
 table(rowSums(ad0_s) > 28)
 
 #Percentage in sequences of Class
pct_meta <- (sum_species["metagenome"]/sum(sum_species))*100
pct_pont <- (sum_species["Pontibacter_populi"]/sum(sum_species))*100
pct_unba <- (sum_species["uncultured_bacterium"]/sum(sum_species))*100
pct_unpla <- (sum_species["uncultured_planctomycete1"]/sum(sum_species))*100  
pct_unid <- (sum_species["unidentified"]/sum(sum_species))*100
pct_unpla2 <- (sum_species["uncultured_Planctomycetales"]/sum(sum_species))*100
pct_unoh <- (sum_species["uncultured_Ohtaekwangia"]/sum(sum_species))*100
pct_unch <- (sum_species["uncultured_Chloroflexi"]/sum(sum_species))*100
pct_unaci <- (sum_species["uncultured_Acidimicrobiia"]/sum(sum_species))*100
pct_unact <- (sum_species["uncultured_actinobacterium"]/sum(sum_species))*100
pct_unbac <- (sum_species["uncultured_bacterium1"]/sum(sum_species))*100
pct_unsoil <- (sum_species["uncultured_soil"]/sum(sum_species))*100
```
