---
title: "GOOD_16S_biodiversity_indexes"
author: "group 2 microbiome"
date: "17/6/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(openxlsx)
library(ggplot2)
library(vegan)
library(readr)
library(dplyr)
library(fossil)
```

```{r import tables}
df_phylum <- read.xlsx("16S_phylum_reads.xlsx")
df_order <- read.xlsx("16S_order_reads.xlsx")
df_family <- read.xlsx("16S_family_reads.xlsx")
df_gender <- read.xlsx("16S_gender_reads.xlsx")

# Import the S1 table

S1_table <- read.xlsx("S1_table.xlsx")
colnames(S1_table) <- gsub("\\.","", colnames(S1_table))

# Change format to .csv

write.csv(S1_table, file = "S1_table.csv", row.names = TRUE)

#Delete the same samples as 16S

S1_table_16S <- S1_table[-c( 4, 11), ]
```

```{r phylum}
# Phylum of 16S 


# Shannon for Phylum of 16S

shannon_phylum_16S <- plyr::ddply(df_phylum,~Samples,function(x) {
  data.frame(SHANNON=diversity(x[-1], index = "shannon"))
})
write.csv(shannon_phylum_16S, file = "shannon_phylum_16S.csv", row.names = TRUE)


# Simpson for Phylum of 16S

simpson_phylum_16S <- plyr::ddply(df_phylum,~Samples,function(x) {
  data.frame(SIMPSON=diversity(x[-1], index = "simpson"))
})
write.csv(simpson_phylum_16S, file = "simpson_phylum_16S.csv", row.names = TRUE)


# Chao1 for Phylum of 16S

chao_phylum_16S <- plyr::ddply(df_phylum,~Samples,function(x) {
  data.frame(chao1 =chao1(x[-1], taxa.row = FALSE))
})


# ACE for Phylum of 16S
ACE_phylum_16S <- plyr::ddply(df_phylum,~Samples,function(x) {
  data.frame(ACE = ACE(x[-1], taxa.row = FALSE))
})

# Merge biodiversity indexes

bi_phylum_16S <- merge(shannon_phylum_16S, simpson_phylum_16S,  by = "Samples")
bi_phylum_16S <- merge(bi_phylum_16S,chao_phylum_16S, by = "Samples")
bi_phylum_16S <- merge(bi_phylum_16S,ACE_phylum_16S, by = "Samples")


# Export the merged table in excel and .csv

write.xlsx(bi_phylum_16S,"bi_phylum_16S.xlsx")
write.csv(bi_phylum_16S, file = "bi_phylum_16S.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#Merge bi and S1 tables
data_phylum_16S <- merge(bi_phylum_16S, S1_table_16S, by = "Samples")


# Spearman correlation for SHANNON

corr_phy_16S_shann <- cor.test(x=data_phylum_16S$ECe_values, y=data_phylum_16S$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_phy_16S_shann

# Manually calculating

phy.ranked.shann <- data.frame(cbind(rank(data_phylum_16S$ECe_values, ties.method = "average"), rank(data_phylum_16S$SHANNON, ties.method = "average")))

colnames(phy.ranked.shann) <- c("ECe_values", "SHANNON")

rho <- cov(phy.ranked.shann) / (sd(phy.ranked.shann$ECe_values) * sd(phy.ranked.shann$SHANNON))

rho[[2]]

corr_phy_16S_shann$estimate

corr_phy_16S_shann$statistic

corr_phy_16S_shann$p.value


# Spearman correlation for SIMPSON

corr_phy_16S_sim <- cor.test(x=data_phylum_16S$ECe_values, y=data_phylum_16S$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_phy_16S_sim

# Manually calculating

phy.ranked.sim <- data.frame(cbind(rank(data_phylum_16S$ECe_values, ties.method = "average"), rank(data_phylum_16S$SIMPSON, ties.method = "average")))

colnames(phy.ranked.sim) <- c("ECe_values", "SIMPSON")

rho <- cov(phy.ranked.sim) / (sd(phy.ranked.sim$ECe_values) * sd(phy.ranked.sim$SIMPSON))

rho[[2]]

corr_phy_16S_sim$estimate

corr_phy_16S_sim$statistic

corr_phy_16S_sim$p.value



#### Only for bulk samples ####

bi_phylum_16S_B <- merge(shannon_phylum_16S, simpson_phylum_16S, by = "Samples") %>% slice(1:22)

# Export the merged table in excel and .csv

write.xlsx(bi_phylum_16S_B,"bi_phylum_16S_B.xlsx")
write.csv(bi_phylum_16S_B, file = "bi_phylum_16S_B.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#S1 table for bulk samples
S1_table_16S_B <- S1_table_16S %>% slice(1:22)

#Merge bi and S1 tables
data_phylum_16S_B <- merge(bi_phylum_16S_B, S1_table_16S_B, by = "Samples")


# Spearman correlation for SHANNON

corr_phy_16S_shann_B <- cor.test(x=data_phylum_16S_B$ECe_values, y=data_phylum_16S_B$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_phy_16S_shann_B

# Manually calculating

phy.ranked.shann_B <- data.frame(cbind(rank(data_phylum_16S_B$ECe_values, ties.method = "average"), rank(data_phylum_16S_B$SHANNON, ties.method = "average")))

colnames(phy.ranked.shann_B) <- c("ECe_values", "SHANNON")

rho <- cov(phy.ranked.shann_B) / (sd(phy.ranked.shann_B$ECe_values) * sd(phy.ranked.shann_B$SHANNON))

rho[[2]]

corr_phy_16S_shann_B$estimate

corr_phy_16S_shann_B$statistic

corr_phy_16S_shann_B$p.value


# Spearman correlation for SIMPSON

corr_phy_16S_sim_B <- cor.test(x=data_phylum_16S_B$ECe_values, y=data_phylum_16S_B$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_phy_16S_sim_B

# Manually calculating

phy.ranked.sim_B <- data.frame(cbind(rank(data_phylum_16S_B$ECe_values, ties.method = "average"), rank(data_phylum_16S_B$SIMPSON, ties.method = "average")))

colnames(phy.ranked.sim_B) <- c("ECe_values", "SIMPSON")

rho <- cov(phy.ranked.sim_B) / (sd(phy.ranked.sim_B$ECe_values) * sd(phy.ranked.sim_B$SIMPSON))

rho[[2]]

corr_phy_16S_sim_B$estimate

corr_phy_16S_sim_B$statistic

corr_phy_16S_sim_B$p.value


#### Only for rhizospheric samples ####

bi_phylum_16S_R <- merge(shannon_phylum_16S, simpson_phylum_16S, by = "Samples") %>% slice(23:46)

# Export the merged table in excel and .csv

write.xlsx(bi_phylum_16S_R,"bi_phylum_16S_R.xlsx")
write.csv(bi_phylum_16S_R, file = "bi_phylum_16S_R.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#S1 table for rhizosphere samples
S1_table_16S_R <- S1_table_16S %>% slice(23:46)

#Merge bi and S1 tables
data_phylum_16S_R <- merge(bi_phylum_16S_R, S1_table_16S_R, by = "Samples")


# Spearman correlation for SHANNON

corr_phy_16S_shann_R <- cor.test(x=data_phylum_16S_R$ECe_values, y=data_phylum_16S_R$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_phy_16S_shann_R

# Manually calculating

phy.ranked.shann_R <- data.frame(cbind(rank(data_phylum_16S_R$ECe_values, ties.method = "average"), rank(data_phylum_16S_R$SHANNON, ties.method = "average")))

colnames(phy.ranked.shann_R) <- c("ECe_values", "SHANNON")

rho <- cov(phy.ranked.shann_R) / (sd(phy.ranked.shann_R$ECe_values) * sd(phy.ranked.shann_R$SHANNON))

rho[[2]]

corr_phy_16S_shann_R$estimate

corr_phy_16S_shann_R$statistic

corr_phy_16S_shann_R$p.value


# Spearman correlation for SIMPSON

corr_phy_16S_sim_R <- cor.test(x=data_phylum_16S_R$ECe_values, y=data_phylum_16S_R$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_phy_16S_sim_R

# Manually calculating

phy.ranked.sim_R <- data.frame(cbind(rank(data_phylum_16S_R$ECe_values, ties.method = "average"), rank(data_phylum_16S_R$SIMPSON, ties.method = "average")))

colnames(phy.ranked.sim_R) <- c("ECe_values", "SIMPSON")

rho <- cov(phy.ranked.sim_R) / (sd(phy.ranked.sim_R$ECe_values) * sd(phy.ranked.sim_R$SIMPSON))

rho[[2]]

corr_phy_16S_sim_R$estimate

corr_phy_16S_sim_R$statistic

corr_phy_16S_sim_R$p.value



### Kruskal-Wallis test ###

# Phylum 16S SHANNON in bulk samples

kruskal.test(ECe_values~SHANNON, data = phy.ranked.shann_B)

# Phylum 16S SIMPSON in bulk samples

kruskal.test(ECe_values~SIMPSON, data = phy.ranked.sim_B)

# Phylum 16S SHANNON in rhizospheric samples

kruskal.test(ECe_values~SHANNON, data = phy.ranked.shann_R)

# Phylum 16S SIMPSON in rhizospheric samples

kruskal.test(ECe_values~SIMPSON, data = phy.ranked.sim_R)


#### Plots

ggplot(data = phy.ranked.sim_R, mapping =aes(x = ECe_values, y = SIMPSON)) + geom_point() + geom_smooth()

```




```{r order}
# Order of 16S

# Shannon for Order of 16S

shannon_order_16S <- plyr::ddply(df_order,~Samples,function(x) {
  data.frame(SHANNON=diversity(x[-1], index = "shannon"))
})
write.csv(shannon_order_16S, file = "shannon_order_16S.csv", row.names = TRUE)

# Simpson for Order of 16S

simpson_order_16S <- plyr::ddply(df_order,~Samples,function(x) {
  data.frame(SIMPSON=diversity(x[-1], index = "simpson"))
})
write.csv(simpson_order_16S, file = "simpson_order_16S.csv", row.names = TRUE)

# Chao1 for Order of 16S

chao_order_16S <- plyr::ddply(df_order,~Samples,function(x) {
  data.frame(chao1 =chao1(x[-1], taxa.row = FALSE))
})


# ACE for Phylum of 16S
ACE_order_16S <- plyr::ddply(df_order,~Samples,function(x) {
  data.frame(ACE = ACE(x[-1], taxa.row = FALSE))
})

# Merge biodiversity indexes

bi_order_16S <- merge(shannon_order_16S, simpson_order_16S,  by = "Samples")
bi_order_16S <- merge(bi_order_16S,chao_order_16S, by = "Samples")
bi_order_16S <- merge(bi_order_16S,ACE_order_16S, by = "Samples")


# Export the merged table in excel and .csv

write.xlsx(bi_order_16S,"bi_order_16S.xlsx")
write.csv(bi_order_16S, file = "bi_order_16S.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#Merge bi and S1 tables
data_order_16S <- merge(bi_order_16S, S1_table, by = "Samples")


# Spearman correlation for SHANNON

corr_order_16S_shann <- cor.test(x=data_order_16S$ECe_values, y=data_order_16S$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_order_16S_shann

# Manually calculating

order.ranked.shann <- data.frame(cbind(rank(data_order_16S$ECe_values, ties.method = "average"), rank(data_order_16S$SHANNON, ties.method = "average")))

colnames(order.ranked.shann) <- c("ECe_values", "SHANNON")

rho <- cov(order.ranked.shann) / (sd(order.ranked.shann$ECe_values) * sd(order.ranked.shann$SHANNON))

rho[[2]]

corr_order_16S_shann$estimate

corr_order_16S_shann$statistic

corr_order_16S_shann$p.value


# Spearman correlation for SIMPSON

corr_order_16S_sim <- cor.test(x=data_order_16S$ECe_values, y=data_order_16S$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_order_16S_sim

# Manually calculating

order.ranked.sim <- data.frame(cbind(rank(data_order_16S$ECe_values, ties.method = "average"), rank(data_order_16S$SIMPSON, ties.method = "average")))

colnames(order.ranked.sim) <- c("ECe_values", "SIMPSON")

rho <- cov(order.ranked.sim) / (sd(order.ranked.sim$ECe_values) * sd(order.ranked.sim$SIMPSON))

rho[[2]]

corr_order_16S_sim$estimate

corr_order_16S_sim$statistic

corr_order_16S_sim$p.value



#### Only for bulk samples ####

bi_order_16S_B <- merge(shannon_order_16S, simpson_order_16S, by = "Samples") %>% slice(1:22)

# Export the merged table in excel and .csv

write.xlsx(bi_order_16S_B,"bi_order_16S_B.xlsx")
write.csv(bi_order_16S_B, file = "bi_order_16S_B.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#S1 table for bulk samples
S1_table_16S_B <- S1_table_16S %>% slice(1:22)

#Merge bi and S1 tables
data_order_16S_B <- merge(bi_order_16S_B, S1_table_16S_B, by = "Samples")


# Spearman correlation for SHANNON

corr_ord_16S_shann_B <- cor.test(x=data_order_16S_B$ECe_values, y=data_order_16S_B$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_ord_16S_shann_B

# Manually calculating

ord.ranked.shann_B <- data.frame(cbind(rank(data_order_16S_B$ECe_values, ties.method = "average"), rank(data_order_16S_B$SHANNON, ties.method = "average")))

colnames(ord.ranked.shann_B) <- c("ECe_values", "SHANNON")

rho <- cov(ord.ranked.shann_B) / (sd(ord.ranked.shann_B$ECe_values) * sd(ord.ranked.shann_B$SHANNON))

rho[[2]]

corr_ord_16S_shann_B$estimate

corr_ord_16S_shann_B$statistic

corr_ord_16S_shann_B$p.value


# Spearman correlation for SIMPSON

corr_ord_16S_sim_B <- cor.test(x=data_order_16S_B$ECe_values, y=data_order_16S_B$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_ord_16S_sim_B

# Manually calculating

ord.ranked.sim_B <- data.frame(cbind(rank(data_order_16S_B$ECe_values, ties.method = "average"), rank(data_order_16S_B$SIMPSON, ties.method = "average")))

colnames(ord.ranked.sim_B) <- c("ECe_values", "SIMPSON")

rho <- cov(ord.ranked.sim_B) / (sd(ord.ranked.sim_B$ECe_values) * sd(ord.ranked.sim_B$SIMPSON))

rho[[2]]

corr_ord_16S_sim_B$estimate

corr_ord_16S_sim_B$statistic

corr_ord_16S_sim_B$p.value


#### Only for rhizospheric samples ####

bi_order_16S_R <- merge(shannon_order_16S, simpson_order_16S, by = "Samples") %>% slice(23:46)
View(bi_order_16S_R)

# Export the merged table in excel and .csv

write.xlsx(bi_order_16S_R,"bi_order_16S_R.xlsx")
write.csv(bi_order_16S_R, file = "bi_order_16S_R.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#S1 table for rhizosphere samples
S1_table_16S_R <- S1_table_16S %>% slice(23:46)

#Merge bi and S1 tables
data_order_16S_R <- merge(bi_order_16S_R, S1_table_16S_R, by = "Samples")


# Spearman correlation for SHANNON

corr_ord_16S_shann_R <- cor.test(x=data_order_16S_R$ECe_values, y=data_order_16S_R$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_ord_16S_shann_R

# Manually calculating

ord.ranked.shann_R <- data.frame(cbind(rank(data_order_16S_R$ECe_values, ties.method = "average"), rank(data_order_16S_R$SHANNON, ties.method = "average")))

colnames(ord.ranked.shann_R) <- c("ECe_values", "SHANNON")

rho <- cov(ord.ranked.shann_R) / (sd(ord.ranked.shann_R$ECe_values) * sd(ord.ranked.shann_R$SHANNON))

rho[[2]]

corr_ord_16S_shann_R$estimate

corr_ord_16S_shann_R$statistic

corr_ord_16S_shann_R$p.value


# Spearman correlation for SIMPSON

corr_ord_16S_sim_R <- cor.test(x=data_order_16S_R$ECe_values, y=data_order_16S_R$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_ord_16S_sim_R

# Manually calculating

ord.ranked.sim_R <- data.frame(cbind(rank(data_order_16S_R$ECe_values, ties.method = "average"), rank(data_order_16S_R$SIMPSON, ties.method = "average")))

colnames(ord.ranked.sim_R) <- c("ECe_values", "SIMPSON")

rho <- cov(ord.ranked.sim_R) / (sd(ord.ranked.sim_R$ECe_values) * sd(ord.ranked.sim_R$SIMPSON))

rho[[2]]

corr_ord_16S_sim_R$estimate

corr_ord_16S_sim_R$statistic

corr_ord_16S_sim_R$p.value



### Kruskal-Wallis test ###

# Order 16S SHANNON in bulk samples

kruskal.test(ECe_values~SHANNON, data = ord.ranked.shann_B)

# Order 16S SIMPSON in bulk samples

kruskal.test(ECe_values~SIMPSON, data = ord.ranked.sim_B)

# Order 16S SHANNON in rhizospheric samples

kruskal.test(ECe_values~SHANNON, data = ord.ranked.shann_R)

# Order 16S SIMPSON in rhizospheric samples

kruskal.test(ECe_values~SIMPSON, data = ord.ranked.sim_R)
```

```{r family}
# Family of 16S

# Shannon for Family of 16S

shannon_family_16S <- plyr::ddply(df_family,~Samples,function(x) {
  data.frame(SHANNON=diversity(x[-1], index = "shannon"))
})
write.csv(shannon_family_16S, file = "shannon_family_16S.csv", row.names = TRUE)

# Simpson for Family of 16S

simpson_family_16S <- plyr::ddply(df_family,~Samples,function(x) {
  data.frame(SIMPSON=diversity(x[-1], index = "simpson"))
})
write.csv(simpson_family_16S, file = "simpson_family_16S.csv", row.names = TRUE)

# Chao1 for family of 16S

chao_family_16S <- plyr::ddply(df_family,~Samples,function(x) {
  data.frame(chao1 =chao1(x[-1], taxa.row = FALSE))
})

# ACE for Phylum of ITS2
ACE_family_16S <- plyr::ddply(df_family,~Samples,function(x) {
  data.frame(ACE = ACE(x[-1], taxa.row = FALSE))
})

# Merge biodiversity indexes

bi_family_16S <- merge(shannon_family_16S, simpson_family_16S,  by = "Samples")
bi_family_16S <- merge(bi_family_16S,chao_family_16S, by = "Samples")
bi_family_16S <- merge(bi_family_16S,ACE_family_16S, by = "Samples")


# Export the merged table in excel and .csv

write.xlsx(bi_family_16S,"bi_family_16S.xlsx")
write.csv(bi_family_16S, file = "bi_family_16S.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#Merge bi and S1 tables
data_family_16S <- merge(bi_family_16S, S1_table, by = "Samples")


# Spearman correlation for SHANNON

corr_family_16S_shann <- cor.test(x=data_family_16S$ECe_values, y=data_family_16S$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_family_16S_shann

# Manually calculating

family.ranked.shann <- data.frame(cbind(rank(data_family_16S$ECe_values, ties.method = "average"), rank(data_family_16S$SHANNON, ties.method = "average")))

colnames(family.ranked.shann) <- c("ECe_values", "SHANNON")

rho <- cov(family.ranked.shann) / (sd(family.ranked.shann$ECe_values) * sd(family.ranked.shann$SHANNON))

rho[[2]]

corr_family_16S_shann$estimate

corr_family_16S_shann$statistic

corr_family_16S_shann$p.value


# Spearman correlation for SIMPSON

corr_family_16S_sim <- cor.test(x=data_family_16S$ECe_values, y=data_family_16S$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_family_16S_sim

# Manually calculating

family.ranked.sim <- data.frame(cbind(rank(data_family_16S$ECe_values, ties.method = "average"), rank(data_family_16S$SIMPSON, ties.method = "average")))

colnames(family.ranked.sim) <- c("ECe_values", "SIMPSON")

rho <- cov(family.ranked.sim) / (sd(family.ranked.sim$ECe_values) * sd(family.ranked.sim$SIMPSON))

rho[[2]]

corr_family_16S_sim$estimate

corr_family_16S_sim$statistic

corr_family_16S_sim$p.value



#### Only for bulk samples ####

bi_family_16S_B <- merge(shannon_family_16S, simpson_family_16S, by = "Samples") %>% slice(1:22)

# Export the merged table in excel and .csv

write.xlsx(bi_family_16S_B,"bi_family_16S_B.xlsx")
write.csv(bi_family_16S_B, file = "bi_family_16S_B.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#S1 table for bulk samples
S1_table_16S_B <- S1_table_16S %>% slice(1:22)

#Merge bi and S1 tables
data_family_16S_B <- merge(bi_family_16S_B, S1_table_16S_B, by = "Samples")


# Spearman correlation for SHANNON

corr_fam_16S_shann_B <- cor.test(x=data_family_16S_B$ECe_values, y=data_family_16S_B$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_fam_16S_shann_B

# Manually calculating

fam.ranked.shann_B <- data.frame(cbind(rank(data_family_16S_B$ECe_values, ties.method = "average"), rank(data_family_16S_B$SHANNON, ties.method = "average")))

colnames(fam.ranked.shann_B) <- c("ECe_values", "SHANNON")

rho <- cov(fam.ranked.shann_B) / (sd(fam.ranked.shann_B$ECe_values) * sd(fam.ranked.shann_B$SHANNON))

rho[[2]]

corr_fam_16S_shann_B$estimate

corr_fam_16S_shann_B$statistic

corr_fam_16S_shann_B$p.value


# Spearman correlation for SIMPSON

corr_fam_16S_sim_B <- cor.test(x=data_family_16S_B$ECe_values, y=data_family_16S_B$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_fam_16S_sim_B

# Manually calculating

fam.ranked.sim_B <- data.frame(cbind(rank(data_family_16S_B$ECe_values, ties.method = "average"), rank(data_family_16S_B$SIMPSON, ties.method = "average")))

colnames(fam.ranked.sim_B) <- c("ECe_values", "SIMPSON")

rho <- cov(fam.ranked.sim_B) / (sd(fam.ranked.sim_B$ECe_values) * sd(fam.ranked.sim_B$SIMPSON))

rho[[2]]

corr_fam_16S_sim_B$estimate

corr_fam_16S_sim_B$statistic

corr_fam_16S_sim_B$p.value


#### Only for rhizospheric samples ####

bi_family_16S_R <- merge(shannon_family_16S, simpson_family_16S, by = "Samples") %>% slice(23:46)
View(bi_family_16S_R)

# Export the merged table in excel and .csv

write.xlsx(bi_family_16S_R,"bi_family_16S_R.xlsx")
write.csv(bi_family_16S_R, file = "bi_family_16S_R.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#S1 table for rhizosphere samples
S1_table_16S_R <- S1_table_16S %>% slice(23:46)

#Merge bi and S1 tables
data_family_16S_R <- merge(bi_family_16S_R, S1_table_16S_R, by = "Samples")


# Spearman correlation for SHANNON

corr_fam_16S_shann_R <- cor.test(x=data_family_16S_R$ECe_values, y=data_family_16S_R$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_fam_16S_shann_R

# Manually calculating

fam.ranked.shann_R <- data.frame(cbind(rank(data_family_16S_R$ECe_values, ties.method = "average"), rank(data_family_16S_R$SHANNON, ties.method = "average")))

colnames(fam.ranked.shann_R) <- c("ECe_values", "SHANNON")

rho <- cov(fam.ranked.shann_R) / (sd(fam.ranked.shann_R$ECe_values) * sd(fam.ranked.shann_R$SHANNON))

rho[[2]]

corr_fam_16S_shann_R$estimate

corr_fam_16S_shann_R$statistic

corr_fam_16S_shann_R$p.value


# Spearman correlation for SIMPSON

corr_fam_16S_sim_R <- cor.test(x=data_family_16S_R$ECe_values, y=data_family_16S_R$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_fam_16S_sim_R

# Manually calculating

fam.ranked.sim_R <- data.frame(cbind(rank(data_family_16S_R$ECe_values, ties.method = "average"), rank(data_family_16S_R$SIMPSON, ties.method = "average")))

colnames(fam.ranked.sim_R) <- c("ECe_values", "SIMPSON")

rho <- cov(fam.ranked.sim_R) / (sd(fam.ranked.sim_R$ECe_values) * sd(fam.ranked.sim_R$SIMPSON))

rho[[2]]

corr_fam_16S_sim_R$estimate

corr_fam_16S_sim_R$statistic

corr_fam_16S_sim_R$p.value



### Kruskal-Wallis test ###

# Family 16S SHANNON in bulk samples

kruskal.test(ECe_values~SHANNON, data = fam.ranked.shann_B)

# Family 16S SIMPSON in bulk samples

kruskal.test(ECe_values~SIMPSON, data = fam.ranked.sim_B)

# Family 16S SHANNON in rhizospheric samples

kruskal.test(ECe_values~SHANNON, data = fam.ranked.shann_R)

# Family 16S SIMPSON in rhizospheric samples

kruskal.test(ECe_values~SIMPSON, data = fam.ranked.sim_R)
```

```{r gender}
# Gender of 16S

# Shannon for Gender of 16S

shannon_gender_16S <- plyr::ddply(df_gender,~Samples,function(x) {
  data.frame(SHANNON=diversity(x[-1], index = "shannon"))
})
write.csv(shannon_gender_16S, file = "shannon_gender_16S.csv", row.names = TRUE)

# Simpson for Gender of 16S

simpson_gender_16S <- plyr::ddply(df_gender,~Samples,function(x) {
  data.frame(SIMPSON=diversity(x[-1], index = "simpson"))
})
write.csv(simpson_gender_16S, file = "simpson_gender_16S.csv", row.names = TRUE)

# Chao1 for family of 16S

chao_gender_16S <- plyr::ddply(df_gender,~Samples,function(x) {
  data.frame(chao1 =chao1(x[-1], taxa.row = FALSE))
})

# ACE for Phylum of 16S
ACE_gender_16S <- plyr::ddply(df_gender,~Samples,function(x) {
  data.frame(ACE = ACE(x[-1], taxa.row = FALSE))
})

# Merge biodiversity indexes

bi_gender_16S <- merge(shannon_gender_16S, simpson_gender_16S,  by = "Samples")
bi_gender_16S <- merge(bi_gender_16S,chao_gender_16S, by = "Samples")
bi_gender_16S <- merge(bi_gender_16S,ACE_gender_16S, by = "Samples")


# Export the merged table in excel and .csv

write.xlsx(bi_gender_16S,"bi_gender_16S.xlsx")
write.csv(bi_gender_16S, file = "bi_gender_16S.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#Merge bi and S1 tables
data_gender_16S <- merge(bi_gender_16S, S1_table, by = "Samples")


# Spearman correlation for SHANNON

corr_gender_16S_shann <- cor.test(x=data_gender_16S$ECe_values, y=data_gender_16S$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_gender_16S_shann

# Manually calculating

gender.ranked.shann <- data.frame(cbind(rank(data_gender_16S$ECe_values, ties.method = "average"), rank(data_gender_16S$SHANNON, ties.method = "average")))

colnames(gender.ranked.shann) <- c("ECe_values", "SHANNON")

rho <- cov(gender.ranked.shann) / (sd(gender.ranked.shann$ECe_values) * sd(gender.ranked.shann$SHANNON))

rho[[2]]

corr_gender_16S_shann$estimate

corr_gender_16S_shann$statistic

corr_gender_16S_shann$p.value


# Spearman correlation for SIMPSON

corr_gender_16S_sim <- cor.test(x=data_gender_16S$ECe_values, y=data_gender_16S$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_gender_16S_sim

# Manually calculating

gender.ranked.sim <- data.frame(cbind(rank(data_gender_16S$ECe_values, ties.method = "average"), rank(data_gender_16S$SIMPSON, ties.method = "average")))

colnames(gender.ranked.sim) <- c("ECe_values", "SIMPSON")

rho <- cov(gender.ranked.sim) / (sd(gender.ranked.sim$ECe_values) * sd(gender.ranked.sim$SIMPSON))

rho[[2]]

corr_gender_16S_sim$estimate

corr_gender_16S_sim$statistic

corr_gender_16S_sim$p.value



#### Only for bulk samples ####

bi_gender_16S_B <- merge(shannon_gender_16S, simpson_gender_16S, by = "Samples") %>% slice(1:22)

# Export the merged table in excel and .csv

write.xlsx(bi_gender_16S_B,"bi_gender_16S_B.xlsx")
write.csv(bi_gender_16S_B, file = "bi_gender_16S_B.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#S1 table for bulk samples
S1_table_16S_B <- S1_table_16S %>% slice(1:22)

#Merge bi and S1 tables
data_gender_16S_B <- merge(bi_gender_16S_B, S1_table_16S_B, by = "Samples")


# Spearman correlation for SHANNON

corr_gen_16S_shann_B <- cor.test(x=data_gender_16S_B$ECe_values, y=data_gender_16S_B$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_gen_16S_shann_B

# Manually calculating

gen.ranked.shann_B <- data.frame(cbind(rank(data_gender_16S_B$ECe_values, ties.method = "average"), rank(data_gender_16S_B$SHANNON, ties.method = "average")))

colnames(gen.ranked.shann_B) <- c("ECe_values", "SHANNON")

rho <- cov(gen.ranked.shann_B) / (sd(gen.ranked.shann_B$ECe_values) * sd(gen.ranked.shann_B$SHANNON))

rho[[2]]

corr_gen_16S_shann_B$estimate

corr_gen_16S_shann_B$statistic

corr_gen_16S_shann_B$p.value


# Spearman correlation for SIMPSON

corr_gen_16S_sim_B <- cor.test(x=data_gender_16S_B$ECe_values, y=data_gender_16S_B$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_gen_16S_sim_B

# Manually calculating

gen.ranked.sim_B <- data.frame(cbind(rank(data_gender_16S_B$ECe_values, ties.method = "average"), rank(data_gender_16S_B$SIMPSON, ties.method = "average")))

colnames(gen.ranked.sim_B) <- c("ECe_values", "SIMPSON")

rho <- cov(gen.ranked.sim_B) / (sd(gen.ranked.sim_B$ECe_values) * sd(gen.ranked.sim_B$SIMPSON))

rho[[2]]

corr_gen_16S_sim_B$estimate

corr_gen_16S_sim_B$statistic

corr_gen_16S_sim_B$p.value


#### Only for rhizospheric samples ####

bi_gender_16S_R <- merge(shannon_gender_16S, simpson_gender_16S, by = "Samples") %>% slice(23:46)
View(bi_gender_16S_R)

# Export the merged table in excel and .csv

write.xlsx(bi_gender_16S_R,"bi_gender_16S_R.xlsx")
write.csv(bi_gender_16S_R, file = "bi_gender_16S_R.csv", row.names = TRUE)


# Preparation of data for Spearman correlation

#S1 table for rhizosphere samples
S1_table_16S_R <- S1_table_16S %>% slice(23:46)

#Merge bi and S1 tables
data_gender_16S_R <- merge(bi_gender_16S_R, S1_table_16S_R, by = "Samples")


# Spearman correlation for SHANNON

corr_gen_16S_shann_R <- cor.test(x=data_gender_16S_R$ECe_values, y=data_gender_16S_R$SHANNON, method = "spearman")

#Cannot compute exact p-value with ties

corr_gen_16S_shann_R

# Manually calculating

gen.ranked.shann_R <- data.frame(cbind(rank(data_gender_16S_R$ECe_values, ties.method = "average"), rank(data_gender_16S_R$SHANNON, ties.method = "average")))

colnames(gen.ranked.shann_R) <- c("ECe_values", "SHANNON")

rho <- cov(gen.ranked.shann_R) / (sd(gen.ranked.shann_R$ECe_values) * sd(gen.ranked.shann_R$SHANNON))

rho[[2]]

corr_gen_16S_shann_R$estimate

corr_gen_16S_shann_R$statistic

corr_gen_16S_shann_R$p.value


# Spearman correlation for SIMPSON

corr_gen_16S_sim_R <- cor.test(x=data_gender_16S_R$ECe_values, y=data_gender_16S_R$SIMPSON, method = "spearman")

#Cannot compute exact p-value with ties

corr_gen_16S_sim_R

# Manually calculating

gen.ranked.sim_R <- data.frame(cbind(rank(data_gender_16S_R$ECe_values, ties.method = "average"), rank(data_gender_16S_R$SIMPSON, ties.method = "average")))

colnames(gen.ranked.sim_R) <- c("ECe_values", "SIMPSON")

rho <- cov(gen.ranked.sim_R) / (sd(gen.ranked.sim_R$ECe_values) * sd(gen.ranked.sim_R$SIMPSON))

rho[[2]]

corr_gen_16S_sim_R$estimate

corr_gen_16S_sim_R$statistic

corr_gen_16S_sim_R$p.value



### Kruskal-Wallis test ###

# Gender 16S SHANNON in bulk samples

kruskal.test(ECe_values~SHANNON, data = gen.ranked.shann_B)

# Gender 16S SIMPSON in bulk samples

kruskal.test(ECe_values~SIMPSON, data = gen.ranked.sim_B)

# Gender 16S SHANNON in rhizospheric samples

kruskal.test(ECe_values~SHANNON, data = gen.ranked.shann_R)

# Gender 16S SIMPSON in rhizospheric samples

kruskal.test(ECe_values~SIMPSON, data = gen.ranked.sim_R)
```


```{r otu}

#Import the rarefied table for 16S
otu <- read.xlsx("16S_rarefied_table.xlsx")
otu <- mutate(otu, Taxonomy = paste(otu$Domain, otu$Phylum, otu$Class, otu$Order, otu$Family, otu$Gender, otu$Species, sep =";")) %>% 
  select(-Domain, -Phylum, -Class, -Order, -Family, -Gender, -Species)
otu <- otu %>% select(Taxonomy, everything())

#Transpose the table
t_otu <- data.frame(t(otu[-1]))
colnames(t_otu) <- otu$Taxonomy
t_otu <- rownames_to_column(t_otu, "Samples")

# Shannon for OTU of 16S
shannon_otu_16S <- plyr::ddply(t_otu,~Samples,function(x) {
  data.frame(SHANNON=diversity(x[-1], index = "shannon"))
})

# Simpson for OTU of 16S

simpson_otu_16S <- plyr::ddply(t_otu,~Samples,function(x) {
  data.frame(SIMPSON=diversity(x[-1], index = "simpson"))
})

# Chao1 for OTU of 16S

chao_otu_16S <- plyr::ddply(t_otu,~Samples,function(x) {
  data.frame(chao1 =chao1(x[-1], taxa.row = FALSE))
})

# ACE for OTU of 16S
ACE_otu_16S <- plyr::ddply(t_otu,~Samples,function(x) {
  data.frame(ACE = ACE(x[-1], taxa.row = FALSE))
})

# Merge biodiversity indexes

bi_otu_16S <- merge(shannon_otu_16S, simpson_otu_16S,  by = "Samples")
bi_otu_16S <- merge(bi_otu_16S,chao_otu_16S, by = "Samples")
bi_otu_16S <- merge(bi_otu_16S,ACE_otu_16S, by = "Samples")


# Export the merged table in excel and .csv

write.xlsx(bi_otu_16S,"bi_otu_16S.xlsx")
write.csv(bi_otu_16S, file = "bi_otu_16S.csv", row.names = TRUE)

```

